contracts/contracts/vault/OETHBaseVaultCore.sol

```diff
@@ -1,12 +1,96 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import { StableMath } from "../utils/StableMath.sol";
 import { OETHVaultCore } from "./OETHVaultCore.sol";
 
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IStrategy } from "../interfaces/IStrategy.sol";
+
 /**
  * @title OETH Base VaultCore Contract
  * @author Origin Protocol Inc
  */
 contract OETHBaseVaultCore is OETHVaultCore {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
     constructor(address _weth) OETHVaultCore(_weth) {}
+
+    /**
+     * @notice Mints Token on behalf of the strategy
+     *          and send it to receiver address
+     * @param receiver Address of the receiver
+     * @param amount Amount of OTokens to mint
+     */
+    function _mintToForStrategy(address receiver, uint256 amount)
+        internal
+        whenNotCapitalPaused
+    {
+        require(
+            strategies[msg.sender].isSupported == true,
+            "Unsupported strategy"
+        );
+        require(
+            mintWhitelistedStrategy[msg.sender] == true,
+            "Not whitelisted strategy"
+        );
+
+        require(amount < MAX_INT, "Amount too high");
+
+        emit Mint(receiver, amount);
+
+        // Mint matching amount of OTokens
+        oUSD.mint(receiver, amount);
+    }
+
+    // @inheritdoc VaultCore
+    function mintForStrategy(uint256 amount) external override {
+        _mintToForStrategy(msg.sender, amount);
+    }
+
+    // Same as mintForStrategy(uint256) but transfers
+    // minted tokens to a different address
+    function mintToForStrategy(address receiver, uint256 amount) external {
```

Comment:
Just curious do we need this functions where strategy contracts would mint/burn the tokens and send them directly to another contract?

I am thinking if is going to be confusing and potentially hard to track the tokens if they are not sent to strategy contracts but directly to another address

Reply:
I was thinking of avoiding approve calls within the contract for OETHb. If we remove these functions, that's one additional transfer for mints. And one additional approve (from the caller) and transfer for burns.

> I am thinking if is going to be confusing and potentially hard to track the tokens if they are not sent to strategy contracts but directly to another address

That's a good point, so all printed OETHb tokens have to go through the Vault? Will make that change

Reply:
I was thinking that `mintForStrategy` is going to be called by the strategy contract. And similarly how the function checks if `msg.sender` is the approved strategy (meaning strategy contract is the caller) I was thinking it could also send the minted tokens (and burn excess tokens) directly from/to the strategy contract as well.

Reply:
Ah that is what you meant by another approve and transfer, I see. 

Yup agreed, might be better to keep the Vault code cleaner.